cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1076,6 +1077,7 @@ func NewBicepProvider(
 	infraOptions Options,
 	commandRunner exec.CommandRunner,
 	console input.Console,
+	prompters Prompters,
```

Comment:
I'm not a fan of all these prompts leaking into the business logic but this is better than calling a global utility method.

Reply:
I think the reality is that infra providers either need to prompt, or we need to revisit that design so that parameters are gathered up-front and otherwise fail. But like you said, this is a step up and a refactoring exercise.